core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5117,6 +5123,47 @@ private static void computeVersion(ServletContext context) {
         }
     }
 
+    /**
+     * The default set of permissions for a new file on the platform (computed once at Jenkins startup).
+     * (provides for instance an indirect access to <code>umask</code> for Unices).
+     */
+    @Nullable
+    public static final Set<PosixFilePermission> DEFAULT_FILE_PERMISSIONS;
+
+    static {
+        LOGGER.log(Level.INFO, "Computing default file permissions");
```

Comment:
I'm not convinced that this is necessary. `Files#createFile` calls `Files#newByteChannel`,
and [UnixFileSystemProvider#newByteChannel](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/d5a00b1e8f78/src/solaris/classes/sun/nio/fs/UnixFileSystemProvider.java#l203) uses [UnixFileModeAttribute#ALL_READWRITE](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/d5a00b1e8f78/src/solaris/classes/sun/nio/fs/UnixFileModeAttribute.java#l37) (666) on the new file by default. `Files#createTempFile` calls `TempFileHelper#create` which calls `File#createFile` and then `Files#newByteChannel` but it passes permissions 600 which are used instead of the default. We could just past 666 permissions to `Files#createTempFile` and the code path would be the same (for java versions built from openjdk).

Maybe there is a valid reason to do the default check as it is here if we are concerned that another supported java implementation does something differently? The `Setting Initial Permissions` section of the documentation for [PosixFileAttributeView](https://docs.oracle.com/javase/7/docs/api/java/nio/file/attribute/PosixFileAttributeView.html) mentions that umask might apply to newly created files, but I don't know if that is enough documentation to be confident in the behavior.